Source/CarthageKit/VersionFile.swift

```diff
@@ -216,47 +237,42 @@ public func createVersionFileForCommitish(
 	buildProducts: [URL],
 	rootDirectoryURL: URL
 ) -> SignalProducer<(), CarthageError> {
-	var platformCaches: [String: [CachedFramework]] = [:]
+	typealias PlatformCaches = [String:[CachedFramework]]
+
+	var platformCaches: PlatformCaches = [:]
 
 	let platformsToCache = platforms.isEmpty ? Set(Platform.supportedPlatforms) : platforms
 	for platform in platformsToCache {
 		platformCaches[platform.rawValue] = []
 	}
 
-	let writeVersionFile = SignalProducer<(), CarthageError> { () -> Result<(), CarthageError> in
-		let rootBinariesURL = rootDirectoryURL.appendingPathComponent(Constants.binariesFolderPath, isDirectory: true).resolvingSymlinksInPath()
-		let versionFileURL = rootBinariesURL.appendingPathComponent(".\(dependencyName).\(VersionFile.pathExtension)")
-
-		let versionFile = VersionFile(
-			commitish: commitish,
-			macOS: platformCaches[Platform.macOS.rawValue],
-			iOS: platformCaches[Platform.iOS.rawValue],
-			watchOS: platformCaches[Platform.watchOS.rawValue],
-			tvOS: platformCaches[Platform.tvOS.rawValue])
-
-		return versionFile.write(to: versionFileURL)
-	}
-
 	if !buildProducts.isEmpty {
 		return SignalProducer<URL, CarthageError>(buildProducts)
-			.flatMap(.merge) { url -> SignalProducer<String, CarthageError> in
-				let platformName = url.deletingLastPathComponent().lastPathComponent
+			.flatMap(.merge) { url -> SignalProducer<(String, (String, String)), CarthageError> in
 				let frameworkName = url.deletingPathExtension().lastPathComponent
+				let platformName = url.deletingLastPathComponent().lastPathComponent
 				let frameworkURL = url.appendingPathComponent(frameworkName, isDirectory: false)
-				return hashForFileAtURL(frameworkURL)
-					.on(value: { hash in
-						let cachedFramework = CachedFramework(name: frameworkName, hash: hash)
-						if var frameworks = platformCaches[platformName] {
-							frameworks.append(cachedFramework)
-							platformCaches[platformName] = frameworks
-						}
-					})
+				let details = SignalProducer<(String, String), CarthageError>(value: (platformName, frameworkName))
+				return SignalProducer.zip(hashForFileAtURL(frameworkURL), details)
+			}
+			.reduce(into: platformCaches) { (platformCaches: inout PlatformCaches, values: (String, (String, String))) in
+				let hash = values.0
+				let platformName = values.1.0
+				let frameworkName = values.1.1
+
+				let cachedFramework = CachedFramework(name: frameworkName, hash: hash)
+				if var frameworks = platformCaches[platformName] {
+					frameworks.append(cachedFramework)
+					platformCaches[platformName] = frameworks
+				}
```

Comment:
This seems like it needs an `else` case to create the array for the first time?

Reply:
As I understand in the code above (https://github.com/Carthage/Carthage/pull/2376/files#diff-6206d1bf8ed3195b3b646bc3d2b6553dR244) it creates empty arrays for each platform we support. Then for platforms that are not supported nothing gets added to the dict. Not sure if that is the behaviour that we need, but this was the way original code was structured.

Reply:
Oops! You're totally right. I was looking for code like that and just couldn't find it.